### DISCUSSION TITLE
Difference between `yay --cleanafter` and `yaycache -r --remove-build-files`

### DISCUSSION THREAD

**Discussion by KAGEYAM4** [aokellermann/yaycache#7] [Q&A] [answered]
Sorry about frequently asking questions.

So yay downloads build-files which can be either:
- prebuild binary in the form of tar.gz
- git repo
- If it is tar.gz it generates pkg.tar.zst which is unpacked and installed.
- If it is git repo, it is compiled, and pkg.tar.zst is generated which is unpacked and installed.

So anything except `pkg.tar.zst` is build files.

`yaycache -r --remove-build-files` - will remove all files except `pkg.tar.zst` ( will consides --keep )
`yay --cleanafter` - what does this do?

**Comment by aokellermann** [comment, marked as answer]
No worries. I just now enabled discussions for this repo so future questions can go here instead: https://github.com/aokellermann/yaycache/discussions

AUR packages are just git repos that usually just contain a `PKGBUILD` and maybe a couple other files. All other files such as the source code's repo and `pkg.tar.zst` are "untracked" in git terms. 

`--remove-build-files`: Removes all untracked files except `pkg.tar.zst` files. It also takes into account arguments like `min-mtime`/`min-atime` and `--ignore`.

`--cleanafter`: According to https://linuxcommandlibrary.com/man/yay it removes all untracked files except directories (no idea why).
